lxd/containers.go

```diff
@@ -1719,6 +1737,16 @@ func (c *lxdContainer) setupMacAddresses(d *Daemon) error {
 			}
 		}
 
+		/*
+		 * propagate the name of the device down for nics if the user
+		 * didn't specify one. This is probably what they wanted
+		 * anyway, and lets us match against network names once they go
+		 * into lxc, e.g. to pull out the veth name for `lxc info`.
+		 */
+		if _, ok := d["name"]; !ok {
+			d["name"] = name
+		}
+
```

Comment:
On Mon, Jun 15, 2015 at 07:44:21AM -0700, Stéphane Graber wrote:
> > @@ -1719,6 +1737,16 @@ func (c *lxdContainer) setupMacAddresses(d *Daemon) error {
> >  			}
> >  		}
> >  
> > +		/*
> > +		 * propagate the name of the device down for nics if the user
> > +		 * didn't specify one. This is probably what they wanted
> > +		 * anyway, and lets us match against network names once they go
> > +		 * into lxc, e.g. to pull out the veth name for `lxc info`.
> > +		 */
> > +		if _, ok := d["name"]; !ok {
> > +			d["name"] = name
> > +		}
> > +
> 
> Just to confirm, "name" there refers to the device name INSIDE the container, correct?

Neither, name here is the key for the devices dict, so:

devices:
  eth0:
    nictype: bridged
    parent: lxcbr0
    type: nic

gets transformed to:

devices:
  eth0:
    nictype: bridged
    parent: lxcbr0
    type: nic
    name: eth0

Which is (I think) a useful transformation, but it's not critical for
this patch any more, so we can drop it if you want.

> If so, then that's fine, if it instead refers to the device name OUTSIDE the container, then we shouldn't do this.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/lxc/lxd/pull/738/files#r32427158